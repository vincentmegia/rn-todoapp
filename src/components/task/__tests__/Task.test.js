import {render} from '@testing-library/react-native'
import Task from '..'
import React from 'react'
import {act} from 'react-test-renderer'

describe('Add components tests', () => {
  const item = {
    key: new Date().getMilliseconds(),
    task: {text: 'write unit test', status: true},
  }

  it('it matches snapshot', () => {
    const {toJSON} = render(
      <Task item={item} onTaskCompleted={jest.fn()} onDelete={jest.fn()} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('user ticks a task complete', () => {
    const {getByTestId} = render(
      <Task item={item} onTaskCompleted={jest.fn()} onDelete={jest.fn()} />,
    )

    act(() => {
      getByTestId('task').props.onClick()
    })

    expect(item.status).toBeFalsy()
  })

  it('user ticks delete', () => {
    const mockedDelete = jest.fn()
    const {getByTestId} = render(
      <Task item={item} onTaskCompleted={jest.fn()} onDelete={mockedDelete} />,
    )

    act(() => {
      getByTestId('delete').props.onClick()
    })

    expect(mockedDelete).toHaveBeenCalledTimes(1)
  })
})
