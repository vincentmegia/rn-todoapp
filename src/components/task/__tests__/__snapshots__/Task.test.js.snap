// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add components tests it matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderRadius": 10,
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "paddingHorizontal": 0,
      "paddingVertical": 0,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": true,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "flex": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "padding": 10,
          "paddingHorizontal": 0,
        }
      }
      testID="task"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#2089dc",
                "fontSize": 24,
              },
              Object {
                "minWidth": 24,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="RNE__Checkbox__Icon"
        >
          
        </Text>
        <Text
          accessibilityRole="text"
          style={
            Object {
              "color": "#43484d",
              "fontWeight": "bold",
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
          testID="RNE__CheckBox__Title"
        >
          write unit test
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginRight": 6,
          "overflow": "hidden",
        }
      }
      testID="RNE__ICON__CONTAINER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="delete"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          testID="RNE__ICON"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#00aced",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="RNE__ICON__Component"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
